### ISSUE TITLE
Uncaught TypeError: Cannot read property 'endIndex' of null

### ISSUE THREAD

**Issue by peranakanfrancis** [atom/atom#18295]
[Enter steps to reproduce:]

1. ...
2. ...

**Atom**: 1.32.0-beta3 x64
**Electron**: 2.0.9
**OS**: Mac OS X 10.14.1
**Thrown From**: Atom Core

### Stack Trace

Uncaught TypeError: Cannot read property 'endIndex' of null

```
At /Applications/Atom Beta.app/Contents/Resources/app/static/<embedded>:11

TypeError: Cannot read property 'endIndex' of null
    at t.exports (/Applications/Atom Beta.app/Contents/Resources/app/static/<embedded>:11:631712)
    at Object.getSuggestions (/Applications/Atom Beta.app/Contents/Resources/app/static/<embedded>:11:159836)
    at n.forEach.t (/Applications/Atom Beta.app/Contents/Resources/app/static/<embedded>:11:642279)
    at Array.forEach (<anonymous>)
    at AutocompleteManager.getSuggestionsFromProviders (/Applications/Atom Beta.app/Contents/Resources/app/static/<embedded>:11:641795)
    at AutocompleteManager.findSuggestions (/Applications/Atom Beta.app/Contents/Resources/app/static/<embedded>:11:641533)
    at AutocompleteManager.requestNewSuggestions (/Applications/Atom Beta.app/Contents/Resources/app/static/<embedded>:11:650063)
    at AutocompleteManager.showOrHideSuggestionListForBufferChanges (/Applications/Atom Beta.app/Contents/Resources/app/static/<embedded>:11:651154)
    at Function.simpleDispatch (/Applications/Atom Beta.app/Contents/Resources/app/static/<embedded>:11:1184010)
    at Emitter.emit (/Applications/Atom Beta.app/Contents/Resources/app/static/<embedded>:11:1185199)
    at TextBuffer.emitDidChangeTextEvent (/Applications/Atom Beta.app/Contents/Resources/app/static/<embedded>:11:499823)
    at TextBuffer.transact (/Applications/Atom Beta.app/Contents/Resources/app/static/<embedded>:11:488081)
    at TextEditor.transact (/Applications/Atom Beta.app/Contents/Resources/app/static/<embedded>:11:24215)
    at mergeIntersectingSelections (/Applications/Atom Beta.app/Contents/Resources/app/static/<embedded>:11:16150)
    at TextEditor.mergeSelections (/Applications/Atom Beta.app/Contents/Resources/app/static/<embedded>:11:39529)
    at TextEditor.mergeIntersectingSelections (/Applications/Atom Beta.app/Contents/Resources/app/static/<embedded>:11:39038)
    at TextEditor.mutateSelectedText (/Applications/Atom Beta.app/Contents/Resources/app/static/<embedded>:11:16113)
    at TextEditor.insertText (/Applications/Atom Beta.app/Contents/Resources/app/static/<embedded>:11:15662)
    at TextEditor.e.(anonymous function) [as insertText] (/Applications/Atom Beta.app/Contents/Resources/app/static/<embedded>:11:255453)
    at TextEditorComponent.didTextInput (/Applications/Atom Beta.app/Contents/Resources/app/static/<embedded>:11:93575)
```

### Commands

```
     -1:40 core:save (input.hidden-input)
  7x -1:38.3.0 core:move-left (input.hidden-input)
  7x -1:37.2.0 core:move-right (input.hidden-input)
     -1:32.7.0 core:move-left (input.hidden-input)
     -1:31.8.0 core:save (input.hidden-input)
     -0:44.6.0 core:move-left (input.hidden-input)
  3x -0:33.2.0 core:move-right (input.hidden-input)
 13x -0:30.2.0 core:backspace (input.hidden-input)
 15x -0:17.5.0 core:move-right (input.hidden-input)
     -0:16.2.0 core:move-down (input.hidden-input)
     -0:15.1.0 core:move-up (input.hidden-input)
     -0:14.7.0 core:move-left (input.hidden-input)
undefined -0:14.4.0 core:move-right (input.hidden-input)
  2x -0:08.9.0 core:move-left (input.hidden-input)
  7x -0:07.0 core:backspace (input.hidden-input)
  2x -0:03.1.0 core:save (input.hidden-input)
```

### Non-Core Packages

```
atom-terminal 0.8.0 
```

**Comment by rsese** [comment]
Thanks for the report!  Can you confirm a few things for us?

* What were you doing at the time of the error?
* Is the error is reproducible?
* If reproducible, do you see the error in safe mode (`atom --safe`)?

**Comment by alicemingay** [comment]
I was writing html code 
I haven't seen it again

**Comment by blsnttt** [comment]
I have run into the same issue today, as soon as I updated to 1.32.0.

I'm working on a Polymer 2 project, I've noticed that the `<style>` tags break the stylesheet syntax color into gray plain text string, as if the whole sheet was commented, in every web component in the project directory.

![sketch 6](https://user-images.githubusercontent.com/7887228/47442007-ce973780-d7b1-11e8-8b98-370505482685.png)

The TypeError is thrown again as soon as I delete and type again the opening `<style>` tag.

**Atom**: 1.32.0 x64
**Electron**: 2.0.9
**OS**: Microsoft Windows 10 Pro
**Thrown From**: Atom Core


### Stack Trace

Uncaught TypeError: Cannot read property 'endIndex' of null

```
At C:\Users\redacted\AppData\Local\atom\app-1.32.0\resources\app\static\<embedded>:11

TypeError: Cannot read property 'endIndex' of null
    at t.exports (~/AppData/Local/atom/app-1.32.0/resources/app/static/<embedded>:11:632572)
    at Object.getSuggestions (~/AppData/Local/atom/app-1.32.0/resources/app/static/<embedded>:11:159836)
    at n.forEach.t (~/AppData/Local/atom/app-1.32.0/resources/app/static/<embedded>:11:643139)
    at Array.forEach (<anonymous>)
    at AutocompleteManager.getSuggestionsFromProviders (~/AppData/Local/atom/app-1.32.0/resources/app/static/<embedded>:11:642655)
    at AutocompleteManager.findSuggestions (~/AppData/Local/atom/app-1.32.0/resources/app/static/<embedded>:11:642393)
    at AutocompleteManager.requestNewSuggestions (~/AppData/Local/atom/app-1.32.0/resources/app/static/<embedded>:11:650923)
    at AutocompleteManager.showOrHideSuggestionListForBufferChanges (~/AppData/Local/atom/app-1.32.0/resources/app/static/<embedded>:11:652014)
    at Function.simpleDispatch (~/AppData/Local/atom/app-1.32.0/resources/app/static/<embedded>:11:1185469)
    at Emitter.emit (~/AppData/Local/atom/app-1.32.0/resources/app/static/<embedded>:11:1186910)
    at TextBuffer.emitDidChangeTextEvent (~/AppData/Local/atom/app-1.32.0/resources/app/static/<embedded>:11:499863)
    at TextBuffer.transact (~/AppData/Local/atom/app-1.32.0/resources/app/static/<embedded>:11:488121)
    at TextEditor.transact (~/AppData/Local/atom/app-1.32.0/resources/app/static/<embedded>:11:24215)
    at mergeIntersectingSelections (~/AppData/Local/atom/app-1.32.0/resources/app/static/<embedded>:11:16150)
    at TextEditor.mergeSelections (~/AppData/Local/atom/app-1.32.0/resources/app/static/<embedded>:11:39529)
    at TextEditor.mergeIntersectingSelections (~/AppData/Local/atom/app-1.32.0/resources/app/static/<embedded>:11:39038)
    at TextEditor.mutateSelectedText (~/AppData/Local/atom/app-1.32.0/resources/app/static/<embedded>:11:16113)
    at TextEditor.insertText (~/AppData/Local/atom/app-1.32.0/resources/app/static/<embedded>:11:15662)
    at TextEditor.e.(anonymous function) [as insertText] (~/AppData/Local/atom/app-1.32.0/resources/app/static/<embedded>:11:255453)
    at TextEditorComponent.didTextInput (~/AppData/Local/atom/app-1.32.0/resources/app/static/<embedded>:11:93575)
```

### Commands

```
     -7:12.6.0 core:undo (input.hidden-input)
     -7:09.5.0 core:backspace (input.hidden-input)
     -7:09.2.0 intentions:highlight (input.hidden-input)
     -7:09.0 core:undo (input.hidden-input)
     -7:05.4.0 intentions:highlight (input.hidden-input)
     -7:05.3.0 core:copy (input.hidden-input)
     -7:02.7.0 editor:newline (input.hidden-input)
     -7:02.4.0 core:move-up (input.hidden-input)
     -7:02.3.0 intentions:highlight (input.hidden-input)
     -7:02.2.0 core:paste (input.hidden-input)
     -7:01.1.0 intentions:highlight (input.hidden-input)
     -7:01.0 core:undo (input.hidden-input)
  3x -6:57.3.0 intentions:highlight (input.hidden-input)
  3x -6:52.7.0 core:undo (input.hidden-input)
     -6:50.6.0 core:delete (input.hidden-input)
  3x -1:18.5.0 core:backspace (input.hidden-input)
```

### Non-Core Packages

```
atom-live-server 2.3.0 
busy-signal 1.4.3 
color-picker 2.3.0 
file-icons 2.1.25 
gzl-syntax 1.0.0 
gzl-ui 3.0.3 
highlight-selected 0.14.0 
intentions 1.1.5 
linter 2.2.0 
linter-ui-default 1.7.1 
minimap 4.29.9 
polymer-snippets 3.2.1 
split-diff 1.5.3 
```

**Comment by kaelifa** [comment]
We're seeing this error to, across multiple machines (Mac and windows) everytime an HTML tag is opened the error is triggered.

**Comment by rsese** [comment]
@blsnttt @kaelifa - can either of you provide a minimal example file with steps to reproduce so we can take a look?

Also, does the error reproduce in [safe mode](https://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode)?

**Comment by rsese** [comment]
@yegor-shuvayev - after that time period, is the error reproducible for you?  Specific steps to reproduce with a minimal example file would be super helpful.

**Comment by repertor** [comment]
I am receiving this error in a new HTML file. I copied plain text into Atom to format with HTML tags, and at the beginning of each tag that I enter the error is triggered. I noticed that the error was triggered as soon as the autocomplete options appear. Disabling autocomplete-plus means I can enter tags without the errors piling up.

**Comment by rsese** [comment]
If you re-enable autocomplete-plus, can you still reproduce in safe mode @repertor?  Also, what version of Atom are you running (`atom -v`) and what operating system?

**Comment by thedorkjedi** [comment]
had it as well:

`
TypeError: Cannot read property 'endIndex' of null
    at t.exports (/Applications/Atom.app/Contents/Resources/app/static/<embedded>:11:634056)
    at Object.getSuggestions (/Applications/Atom.app/Contents/Resources/app/static/<embedded>:11:159840)
    at n.forEach.t (/Applications/Atom.app/Contents/Resources/app/static/<embedded>:11:644623)
    at Array.forEach (<anonymous>)
    at AutocompleteManager.getSuggestionsFromProviders (/Applications/Atom.app/Contents/Resources/app/static/<embedded>:11:644139)
    at AutocompleteManager.findSuggestions (/Applications/Atom.app/Contents/Resources/app/static/<embedded>:11:643877)
    at AutocompleteManager.requestNewSuggestions (/Applications/Atom.app/Contents/Resources/app/static/<embedded>:11:652407)
    at AutocompleteManager.showOrHideSuggestionListForBufferChanges (/Applications/Atom.app/Contents/Resources/app/static/<embedded>:11:653498)
    at Function.simpleDispatch (/Applications/Atom.app/Contents/Resources/app/static/<embedded>:11:1186953)
    at Emitter.emit (/Applications/Atom.app/Contents/Resources/app/static/<embedded>:11:1188394)
    at TextBuffer.emitDidChangeTextEvent (/Applications/Atom.app/Contents/Resources/app/static/<embedded>:11:501342)
    at TextBuffer.transact (/Applications/Atom.app/Contents/Resources/app/static/<embedded>:11:489600)
    at TextEditor.transact (/Applications/Atom.app/Contents/Resources/app/static/<embedded>:11:24215)
    at mergeIntersectingSelections (/Applications/Atom.app/Contents/Resources/app/static/<embedded>:11:16150)
    at TextEditor.mergeSelections (/Applications/Atom.app/Contents/Resources/app/static/<embedded>:11:39484)
    at TextEditor.mergeIntersectingSelections (/Applications/Atom.app/Contents/Resources/app/static/<embedded>:11:39038)
    at TextEditor.mutateSelectedText (/Applications/Atom.app/Contents/Resources/app/static/<embedded>:11:16113)
    at TextEditor.insertText (/Applications/Atom.app/Contents/Resources/app/static/<embedded>:11:15662)
    at TextEditor.e.(anonymous function) [as insertText] (/Applications/Atom.app/Contents/Resources/app/static/<embedded>:11:256932)
    at TextEditorComponent.didTextInput (/Applications/Atom.app/Contents/Resources/app/static/<embedded>:11:93579)
`

it seems like this appears when you make double backslashes then add the text class on html on my part.

It looks like its getting confused by double backslashes.

**Comment by rsese** [comment]
> it seems like this appears when you make double backslashes then add the text class on html on my part.

Thanks @thedorkjedi - can you clarify the step above with an example file and step-by-step instructions to reproduce the error?

**Comment by MeatBaker** [comment]
i copied the error report and tadaaa :

[Enter steps to reproduce:]

1. ...
2. ...

**Atom**: 1.32.2 x64
**Electron**: 2.0.9
**OS**: Unknown Windows version
**Thrown From**: Atom Core


### Stack Trace

Uncaught TypeError: Cannot read property 'endIndex' of null

```
At C:\Users\yohan\AppData\Local\atom\app-1.32.2\resources\app\static\<embedded>:11

TypeError: Cannot read property 'endIndex' of null
    at t.exports (~/AppData/Local/atom/app-1.32.2/resources/app/static/<embedded>:11:634056)
    at Object.getSuggestions (~/AppData/Local/atom/app-1.32.2/resources/app/static/<embedded>:11:159840)
    at n.forEach.t (~/AppData/Local/atom/app-1.32.2/resources/app/static/<embedded>:11:644623)
    at Array.forEach (<anonymous>)
    at AutocompleteManager.getSuggestionsFromProviders (~/AppData/Local/atom/app-1.32.2/resources/app/static/<embedded>:11:644139)
    at AutocompleteManager.findSuggestions (~/AppData/Local/atom/app-1.32.2/resources/app/static/<embedded>:11:643877)
    at AutocompleteManager.requestNewSuggestions (~/AppData/Local/atom/app-1.32.2/resources/app/static/<embedded>:11:652407)
    at AutocompleteManager.showOrHideSuggestionListForBufferChanges (~/AppData/Local/atom/app-1.32.2/resources/app/static/<embedded>:11:653498)
    at Function.simpleDispatch (~/AppData/Local/atom/app-1.32.2/resources/app/static/<embedded>:11:1186984)
    at Emitter.emit (~/AppData/Local/atom/app-1.32.2/resources/app/static/<embedded>:11:1188425)
    at TextBuffer.emitDidChangeTextEvent (~/AppData/Local/atom/app-1.32.2/resources/app/static/<embedded>:11:501342)
    at TextBuffer.transact (~/AppData/Local/atom/app-1.32.2/resources/app/static/<embedded>:11:489600)
    at TextEditor.transact (~/AppData/Local/atom/app-1.32.2/resources/app/static/<embedded>:11:24215)
    at mergeIntersectingSelections (~/AppData/Local/atom/app-1.32.2/resources/app/static/<embedded>:11:16150)
    at TextEditor.mergeSelections (~/AppData/Local/atom/app-1.32.2/resources/app/static/<embedded>:11:39529)
    at TextEditor.mergeIntersectingSelections (~/AppData/Local/atom/app-1.32.2/resources/app/static/<embedded>:11:39038)
    at TextEditor.mutateSelectedText (~/AppData/Local/atom/app-1.32.2/resources/app/static/<embedded>:11:16113)
    at TextEditor.insertText (~/AppData/Local/atom/app-1.32.2/resources/app/static/<embedded>:11:15662)
    at TextEditor.e.(anonymous function) [as insertText] (~/AppData/Local/atom/app-1.32.2/resources/app/static/<embedded>:11:256932)
    at TextEditorComponent.didTextInput (~/AppData/Local/atom/app-1.32.2/resources/app/static/<embedded>:11:93579)
```

### Commands

```
     -7:06.7.0 autocomplete-plus:confirm (input.hidden-input)
  9x -7:05.5.0 core:backspace (input.hidden-input)
     -7:02.0 autocomplete-plus:confirm (input.hidden-input)
     -7:01.9.0 autocomplete-plus:activate (atom-text-editor.editor.is-focused)
     -6:43.3.0 core:save (input.hidden-input)
  6x -6:29.5.0 core:move-left (input.hidden-input)
  7x -6:28 core:backspace (input.hidden-input)
     -6:25.7.0 core:save (input.hidden-input)
  5x -6:15.3.0 core:undo (input.hidden-input)
 42x -6:05.2.0 core:backspace (input.hidden-input)
     -5:53 core:save (input.hidden-input)
  4x -5:20.6.0 core:move-left (input.hidden-input)
  2x -5:18.7.0 core:backspace (input.hidden-input)
     -5:17.7.0 core:save (input.hidden-input)
     -4:24.6.0 autocomplete-plus:confirm (input.hidden-input)
     -4:24.6.0 autocomplete-plus:activate (atom-text-editor.editor.is-focused)
```

### Non-Core Packages

```

```

i got this error when i tried to add a class="" to a <td>.

**Comment by LukasPtakauskas** [comment]
Uncaught TypeError: Cannot read property 'endIndex' of null I've got this error while editing html.  
The error popped up every time I started typing `<b` or `<br` in a `<code>` tag.   

`<code>  
      &lt?php<b  
      echo "Hello world";  
      // ... more code  
      echo "Last statement";  
      // the script ends here with no PHP closing tag  
    </code>`  
here's a stacktrace  
`snap/atom/206/usr/share/atom/resources/app/static/<embedded>:11
Hide Stack Trace  
TypeError: Cannot read property 'endIndex' of null  
    at t.exports (/snap/atom/206/usr/share/atom/resources/app/static/<embedded>:11:633245)  
    at Object.getSuggestions (/snap/atom/206/usr/share/atom/resources/app/static/<embedded>:11:159840)  
    at n.forEach.t (/snap/atom/206/usr/share/atom/resources/app/static/<embedded>:11:643812)  
    at Array.forEach (<anonymous>)  
    at AutocompleteManager.getSuggestionsFromProviders (/snap/atom/206/usr/share/atom/resources/app/static/<embedded>:11:643328)  
    at AutocompleteManager.findSuggestions (/snap/atom/206/usr/share/atom/resources/app/static/<embedded>:11:643066)  
    at AutocompleteManager.requestNewSuggestions (/snap/atom/206/usr/share/atom/resources/app/static/<embedded>:11:651596)  
    at AutocompleteManager.showOrHideSuggestionListForBufferChanges (/snap/atom/206/usr/share/atom/resources/app/static/<embedded>:11:652687)  
    at Function.simpleDispatch (/snap/atom/206/usr/share/atom/resources/app/static/<embedded>:11:1186173)  
    at Emitter.emit (/snap/atom/206/usr/share/atom/resources/app/static/<embedded>:11:1187614)  
    at TextBuffer.emitDidChangeTextEvent (/snap/atom/206/usr/share/atom/resources/app/static/<embedded>:11:501342)  
    at TextBuffer.transact (/snap/atom/206/usr/share/atom/resources/app/static/<embedded>:11:489600)  
    at TextEditor.transact (/snap/atom/206/usr/share/atom/resources/app/static/<embedded>:11:24215)  
    at mergeIntersectingSelections (/snap/atom/206/usr/share/atom/resources/app/static/<embedded>:11:16150)  
    at TextEditor.mergeSelections (/snap/atom/206/usr/share/atom/resources/app/static/<embedded>:11:39484)  
    at TextEditor.mergeIntersectingSelections (/snap/atom/206/usr/share/atom/resources/app/static/<embedded>:11:39038)  
    at TextEditor.mutateSelectedText (/snap/atom/206/usr/share/atom/resources/app/static/<embedded>:11:16113)  
    at TextEditor.insertText (/snap/atom/206/usr/share/atom/resources/app/static/<embedded>:11:15662)  
    at TextEditor.e.(anonymous function) [as insertText] (/snap/atom/206/usr/share/atom/resources/app/static/<embedded>:11:256932)  
    at TextEditorComponent.didTextInput (/snap/atom/206/usr/share/atom/resources/app/static/<embedded>:11:93579)`

**Comment by rsese** [comment]
@LukasPtakauskas - I think the fix won't be available till 1.34 but additionally, it looks like you're using the Snap install which isn't an official installation method.   If you can still reproduce with 1.34 when that's released using an [official install](https://flight-manual.atom.io/getting-started/sections/installing-atom/), please open a new issue with the template filled out and we'll take a look.
